compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1539,6 +1539,9 @@ supported_targets! {
     ("aarch64-apple-watchos", aarch64_apple_watchos),
     ("aarch64-apple-watchos-sim", aarch64_apple_watchos_sim),
 
+    ("aarch64-apple-xros", aarch64_apple_xros),
+    ("aarch64-apple-xros-sim", aarch64_apple_xros_sim),
```

Comment:
I disagree that we should use `xrOS` as the name of the target os, even though that's what's used in LLVM; call it `visionOS`, much more intuitive for users!

Previous discussion on this: https://github.com/rust-lang/compiler-team/issues/642#issuecomment-1700922668, opening this thread on the PR so that we can discuss things here.

Reply:
I personally think we should follow what Apple and LLVM does. While it's unfortunate, the xrOS codename is all over Xcode and the platform infrastructure, and I think we should follow that.

Ultimately it doesn't really matter though.

Reply:
If this was only a decision about the target name, then I could live with it, but since it also affects the value of `target_os`, I think it _does_ actually matter quite a lot for code readability!

Compare the following three snippets in Swift, Objective-C and Rust that `cfg`-guards based on running on visionOS.

```swift
// Swift
#if os(visionOS)
print("visionOS-specific code path")
#endif
```

```objective-c
// Objective-C
#if TARGET_OS_VISION
NSLog(@"visionOS-specific code path");
#endif
```

```rust
// Rust
#[cfg(target_os = "xros")]
println!("visionOS-specific code path");
```

Reply:
It's unfortunate that apple has been so inconsistent with the naming of visionOS/xrOS across a lot of the tooling.

Listing of the apple simulators uses `visionOS` (`xcrun simctl list devices xr` yields nothing):
```bash
$ xcrun simctl list devices visionos
== Devices ==
-- iOS 17.2 --
-- tvOS 17.2 --
-- watchOS 10.2 --
-- visionOS 1.0 --
    Apple Vision Pro (B6F630A6-67E2-4322-84BF-F4644F635457) (Shutdown)
```

Yet, the sdk name uses `xros` and `xrsimulator` rather than `visionos` or `visionsimulator`. 

```bash
$ xcrun --show-sdk-path --sdk xrsimulator
/Applications/Xcode.app/Contents/Developer/Platforms/XRSimulator.platform/Developer/SDKs/XRSimulator1.0.sdk
$ xcrun --show-sdk-path --sdk xros
/Applications/Xcode.app/Contents/Developer/Platforms/XROS.platform/Developer/SDKs/XROS1.0.sdk
```

In Swift itself, specifying [supported platforms in a swift package](https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#supportedplatform), it is [`.visionOS("1.0")`](https://github.com/apple/swift-package-manager/blob/d5787ea534ead1cb9a770db03da7cc065df4751a/Sources/PackageModel/Platform.swift#L45). They use [`xros` in internally](https://github.com/apple/swift-package-manager/blob/d5787ea534ead1cb9a770db03da7cc065df4751a/Sources/PackageModel/MinimumDeploymentTarget.swift#L101-L102).

There are more (comical) examples of this inconsistency (`xcrun simctl list runtimes vision` and `xcodebuild -showsdks`) but showing them doesn't add new arguments.

Not that it matters that much but the [zig folks have decided to go with `visionos`](https://github.com/ziglang/zig/issues/18512#issuecomment-1899153738). This is relevant because there is [a pattern to use zig's CC/CXX compiler to cross compile C/C++ dependencies](https://actually.fyi/posts/zig-makes-rust-cross-compilation-just-work/).

Given that Apple is trying to make everything externally be, `visionOS`, I think it'd be less confusing to go with `visionos`.

From a specification perspective, [rust target triples](https://doc.rust-lang.org/nightly/rustc/platform-support.html) use `<architecture>.<vendor>.<os>[<abi>]` - `x86_64-unknown-linux`, `aarch64-apple-darwin`, `arch64-apple-ios-sim`, etc. While `xrOS` is currently mentioned under the History, I don't foresee [the visionOS wikipedia page being renamed to xrOS](https://en.wikipedia.org/wiki/VisionOS).

Reply:
It sounds like the people want `visionOS`. I am extremely busy at the moment, but I will make this change (and the according change in `libc`) when I am available.

Reply:
@bcardarella Apple will likely never clean up all the xrOS references. As an example from their TargetConditionals.h TARGET_OS_MAC is inclusive of the macOS, iOS, iPadOS, tvOS, MacCatalyst, watchOS, visionOS, and the TouchBar on certain MacBook Pros. It definitely confuses people to learn that if they want solely the macOS piece of that the value is TARGET_OS_OSX. For clarity I think it makes sense to go with visionOS here though.